### ISSUE TITLE
[Parity Testing] Remove Silent Fallback to Mock Model in eval_logits_once - Improve Error Visibility for Model Loading Failures

### ISSUE THREAD

**Issue by EffortlessSteven** [EffortlessMetrics/BitNet-rs#285]
## Problem Description

The `eval_logits_once` function in `crates/bitnet-inference/src/parity.rs` has a problematic silent fallback mechanism that creates a mock model when the real model fails to load. This behavior masks actual model loading errors and can hide critical issues during cross-validation testing.

## Current Behavior

**File:** `crates/bitnet-inference/src/parity.rs`
**Function:** `eval_logits_once` (lines 26-60)

```rust
pub fn eval_logits_once(model_path: &str, tokens: &[i32]) -> Result<Vec<f32>> {
    // Try to load model tensors; fall back to a mock model if unavailable
    let (config, model) = match load_gguf(Path::new(model_path), Device::Cpu) {
        Ok((cfg, tensors)) => {
            let model = BitNetModel::from_gguf(cfg.clone(), tensors, Device::Cpu)?;
            (cfg, model)
        }
        Err(_) => {
            let cfg = BitNetConfig::default();
            let model = BitNetModel::new(cfg.clone(), Device::Cpu);
            (cfg, model)
        }
    };
    // ...
}
```

The function silently discards any model loading errors (see `Err(_)` on line 33) and creates a mock model with default configuration. This leads to:

1. **Hidden Errors**: Real model loading failures are completely masked
2. **Invalid Cross-Validation**: Tests may pass with mock data instead of real model outputs  
3. **Debugging Difficulties**: Developers cannot identify why their models fail to load
4. **False Positives**: The function appears to succeed even when the model path is invalid

## Impact Assessment

**Severity:** High - This directly affects the reliability of cross-validation testing
**Affected Components:**
- Cross-validation framework (C++ parity testing)
- Model loading validation 
- Debugging workflows
- Test infrastructure reliability

**Business Impact:**
- Incorrect validation results leading to false confidence in model accuracy
- Wasted developer time debugging issues that are silently masked
- Potential production issues from undetected model loading problems

## Root Cause Analysis

The fallback mechanism was likely implemented to make tests pass in environments without proper model files. However, this approach violates the principle of fail-fast and makes the testing framework unreliable.

The current test at lines 142-158 actually validates this broken behavior:
```rust
#[test]
fn test_mock_eval() {
    // This will use the mock loader since the file doesn't exist
    let result = eval_logits_once("test.gguf", &tokens);
    
    // Should succeed even without a real model file (uses mock)
    assert!(result.is_ok());
}
```

### Primary Implementation

Replace the silent fallback with explicit error propagation:

```rust
pub fn eval_logits_once(model_path: &str, tokens: &[i32]) -> Result<Vec<f32>> {
    let (config, tensors) = load_gguf(Path::new(model_path), Device::Cpu)
        .map_err(|e| anyhow::anyhow!("Failed to load model from '{}': {}", model_path, e))?;

    let model = BitNetModel::from_gguf(config.clone(), tensors, Device::Cpu)?;

    // Convert i32 tokens to u32
    let tokens_u32: Vec<u32> = tokens.iter().map(|&t| t as u32).collect();

    // Create KV cache for the model (batch size 1, CPU)
    let cache = KVCache::new(&config, 1, &candle_core::Device::Cpu)?;
    let mut any_cache: Box<dyn std::any::Any> = Box::new(cache);

    // Get embeddings for the tokens
    let embedded = model.embed(&tokens_u32)?;

    // Run forward pass through the model
    let output = model.forward(&embedded, any_cache.as_mut())?;

    // Get logits from the output
    let logits = model.logits(&output)?;

    // Extract logits for the last token position
    let logits = extract_last_token_logits(logits)?;

    Ok(logits)
}
```

### Alternative Approaches Considered

1. **Environment Variable Toggle**: Add `BITNET_ALLOW_MOCK_FALLBACK` environment variable
   - **Pros**: Backward compatibility for existing tests
   - **Cons**: Still allows problematic behavior, adds complexity

2. **Separate Mock Function**: Create `eval_logits_once_mock()` for explicit mock testing
   - **Pros**: Clear separation of concerns
   - **Cons**: Requires updating all call sites

**Recommendation**: Primary implementation for clean, fail-fast behavior

### Phase 1: Core Fix
- [ ] Remove fallback mechanism in `eval_logits_once` function
- [ ] Update error handling to provide descriptive error messages
- [ ] Remove mock fallback logic (lines 33-37)

### Phase 2: Test Updates  
- [ ] Update `test_mock_eval` test to expect failure with invalid model path
- [ ] Add test for successful model loading with valid GGUF file
- [ ] Add test for descriptive error messages on model loading failure

### Phase 3: Documentation
- [ ] Update function documentation to clarify error behavior
- [ ] Update cross-validation documentation to reflect new error handling

### Phase 4: Related Function Updates
- [ ] Review `eval_logits_incremental` for similar issues
- [ ] Ensure `get_model_vocab_size` and `get_model_config` have proper error handling

### Unit Tests
```rust
#[test]
fn test_eval_logits_once_with_invalid_path() {
    let tokens = vec![1, 2, 3, 4];
    let result = eval_logits_once("nonexistent.gguf", &tokens);
    
    assert!(result.is_err());
    let error_msg = result.unwrap_err().to_string();
    assert!(error_msg.contains("Failed to load model from 'nonexistent.gguf'"));
}

#[test]
fn test_eval_logits_once_with_valid_model() {
    // Requires a test model file
    if let Ok(model_path) = std::env::var("BITNET_TEST_MODEL") {
        let tokens = vec![1, 2, 3, 4];
        let result = eval_logits_once(&model_path, &tokens);
        
        assert!(result.is_ok());
        let logits = result.unwrap();
        assert!(!logits.is_empty());
    }
}
```

### Integration Tests
- Test cross-validation workflow with valid model files
- Verify error messages are helpful for debugging
- Ensure no regression in legitimate model loading scenarios

## Acceptance Criteria

- [ ] `eval_logits_once` returns descriptive errors when model loading fails
- [ ] No silent fallback to mock models occurs
- [ ] All existing tests pass with valid model files
- [ ] Error messages include the failed model path and underlying error
- [ ] Function documentation reflects new error behavior
- [ ] Cross-validation workflows provide clear feedback on model loading issues

## Related Issues

This issue is part of a broader effort to eliminate mock fallbacks and improve error handling:
- **Related**: Issues addressing mock object consolidation (#282)
- **Related**: Production readiness improvements (#278, #283)
- **Blocks**: Reliable cross-validation testing framework
- **Blocked by**: None - can be implemented independently

## Environment Requirements

- **MSRV**: 1.90.0 (Rust 2024 edition)
- **Features**: `--no-default-features --features cpu` for testing
- **Test Dependencies**: Valid GGUF model file for integration tests

## Risk Assessment

**Low Risk**: This change improves error visibility without affecting successful model loading scenarios. The only "breaking" change is that previously silent failures will now be properly reported as errors, which is the desired behavior.

**Migration Path**: Existing code that relied on the fallback behavior will need to handle errors explicitly or provide valid model paths.
